File: 2015/Day01_2015.cpp
#include <iostream>
#include <fstream>

int main()
{
    int floor = 0;
    int position = 0;
    bool setFinal = false;
    int finalPosition = 0;

    std::ifstream file("Day01_2015.txt");
    char c;
    while (file.get(c))
    {
        if (c == '(')
            floor++;
        else if (c == ')')
            floor--;
        position++;
        if (floor == -1 && !setFinal)
        {
            finalPosition = position;
            setFinal = true;
        }
    }

    file.close();
    
    std::cout << "The floor is: " << floor << "\n";
    std::cout << "The position of the first character that enters the basement is: " << finalPosition;

    return 0;
}

File: 2015/Day01_2015.py
floor = 0
position = 0
setFinal = False
finalPosition = 0

with open("Day01_2015.txt") as file:
    for line in file:
        for char in line:
            if char == '(':
                floor += 1
            elif char == ')':
                floor -= 1
            position += 1

            if floor == -1 and not setFinal:
                finalPosition = position
                setFinal = True

file.close()

print("The floor is: " + str(floor))
print("The position of the first character that enters the basement is: " + str(finalPosition))
